### PR TITLE
add support for memorydb

### PR DESCRIPTION
**Pull request by glebpom**
This PR adds support for MemoryDB discovery

### CODE REVIEW

**Review by cristiangreco — COMMENTED** [review]
Thanks for the great contribution @glebpom!!! Would you mind adding a test in the `pkg/job/maxdimassociator` package for this service? You can look at [this PR](https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/937/files) for an example of what needs to be tested.

Once done this is good to be merged!

**Comment by glebpom** [comment]
@cristiangreco I added the test, please check it

File: pkg/job/associator/associator_test.go [hunk 1]
```diff
@@ -68,6 +68,12 @@ var ecsResources = []*model.TaggedResource{
 	ecsService2,
 }
 
+var memoryDBCluster = &model.TaggedResource{
+	ARN:       "arn:aws:memorydb:us-east-1:622820832160:cluster/mycluster",
```
Comment (cristiangreco) [inline pkg/job/associator/associator_test.go:72]:
```suggestion
	ARN:       "arn:aws:memorydb:us-east-1:123456789012:cluster/mycluster",
```

Comment (cristiangreco) [reply pkg/job/associator/associator_test.go:72]: Better to use a placeholder account number.

File: pkg/job/associator/associator_test.go [hunk 2]
```diff
@@ -283,6 +289,25 @@ func TestAssociator(t *testing.T) {
 			expectedSkip:     false,
 			expectedResource: ecsService2,
 		},
+		{
```
Comment (cristiangreco) [inline pkg/job/associator/associator_test.go:292]: Can you please add a similar test case for the "max dimensions associator"? It's a new algorithm currently behind a feature flag, more info [here](https://github.com/nerdswords/yet-another-cloudwatch-exporter/blob/master/docs/feature_flags.md#new-associator-algorithm).

You could create a file named `pkg/job/maxdimassociator/associator_memorydb_test.go` like [this one](https://github.com/nerdswords/yet-another-cloudwatch-exporter/blob/master/pkg/job/maxdimassociator/associator_ec2_test.go).

File: pkg/job/maxdimassociator/associator_memorydb_test.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		{
+			name: "should not skip when unmatching because of non-ARN dimension",
+			args: args{
+				dimensionRegexps: config.SupportedServices.GetService("AWS/MemoryDB").DimensionRegexps,
+				resources:        memoryDBClusters,
+				metric: &model.Metric{
+					Namespace:  "AWS/MemoryDB",
+					MetricName: "BytesUsedForMemoryDB",
+					Dimensions: []*model.Dimension{
+						{Name: "OtherName", Value: "some-other-value"},
```
Comment (glebpom) [inline pkg/job/maxdimassociator/associator_memorydb_test.go:100]: this is copied from EC2, but I don't see any metrics with other dimenstiona

File: pkg/job/associator/associator_test.go (hunk 2 above)
Comment (glebpom) [reply pkg/job/associator/associator_test.go:292]: @cristiangreco added, please take a look

**Review by cristiangreco — APPROVED** [review]
🚀

**Comment by xiaozhuang-a** [comment]
hi,May I ask how to query the resource tag of a specified resource, such as "memorydb: cluster" in memorydb。

**Comment by cristiangreco** [comment]
> hi,May I ask how to query the resource tag of a specified resource, such as "memorydb: cluster" in memorydb。

Have you tried the CLI with something like:

```
aws resourcegroupstaggingapi get-resources --region us-east-1 --resource-type-filters memorydb:cluster 
```

**Pull request merged**